app/src/main/java/com/github/se/travelpouch/model/notifications/NotificationRepositoryFirestore.kt

```diff
@@ -0,0 +1,153 @@
+        .get()
+        .addOnSuccessListener { documents ->
+          for (document in documents) {
+            notificationCollection.document(document.id).delete()
+          }
+          onSuccess()
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  /**
+   * Change the notification type in the Firestore database.
+   *
+   * @param notificationUid The UID of the notification to be changed.
+   * @param notificationType The new type of the notification.
+   */
+  override fun changeNotificationType(notificationUid: String, notificationType: NotificationType) {
+    notificationCollection.document(notificationUid).update("notificationType", notificationType)
+  }
+
+  private fun documentToNotification(document: DocumentSnapshot): Notification {
+    try {
+      val notificationUid = document.id
+      val senderUid = document.getString("senderUid")!!
+      val receiverUid = document.getString("receiverUid")!!
+      val travelUid = document.getString("travelUid")!!
+      val contentData = document["content"] as Map<*, *>
+      val notificationType = NotificationType.valueOf(document.getString("notificationType")!!)
+      val content =
+          when (notificationType) {
+            NotificationType.INVITATION -> {
+              val inviterName = contentData["inviterName"] as String
+              val travelTitle = contentData["travelTitle"] as String
+              val role = contentData["role"] as String
+              NotificationContent.InvitationNotification(
+                  inviterName, travelTitle, Role.valueOf(role))
+            }
+            NotificationType.ROLE_UPDATE -> {
+              val travelTitle = contentData["travelTitle"] as String
+              val role = Role.valueOf(contentData["role"] as String)
+              NotificationContent.RoleChangeNotification(travelTitle, role)
+            }
+            NotificationType.ACCEPTED -> {
+              val userName = contentData["userName"] as String
+              val travelTitle = contentData["travelTitle"] as String
+              NotificationContent.InvitationResponseNotification(userName, travelTitle, true)
+            }
+            NotificationType.DECLINED -> {
+              val userName = contentData["userName"] as String
+              val travelTitle = contentData["travelTitle"] as String
+              NotificationContent.InvitationResponseNotification(userName, travelTitle, false)
+            }
+          }
+      val timestamp = document.getTimestamp("timestamp")!!
+      val status = NotificationStatus.valueOf(document.getString("status")!!)
+
+      return Notification(
+          notificationUid,
+          senderUid,
+          receiverUid,
+          travelUid,
+          content,
+          notificationType,
+          timestamp,
+          status)
+    } catch (e: Exception) {
+      Log.e("NotificationRepository", "Error converting document to Notification", e)
+      throw e
```

Comment:
For coherency accross the project, it may be more interesting to return null and not throwing an exception in case of an error